Missing anymatch dependencies for typing

<!-- Please don't delete this template otherwise your issue will be closed immediately -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

### Bug report

Checking TypeScript types declarations in my project depending on `webpack-dev-server` fails.

Running `tsc`.
<!-- Please ask questions on discussions, StackOverflow or the webpack Gitter. -->
<!-- https://github.com/webpack/webpack/discussions -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

### Actual Behavior

I get this error:
```
node_modules/webpack-dev-server/types/lib/Server.d.ts:3481:37 - error TS7016: Could not find a declaration file for module 'anymatch'. '/usr/app/node_modules/anymatch/index.js' implicitly has an 'any' type.
```

### Expected Behavior

It should not raise an error (probably by pulling the `anymatch` in the dependencies).

### How Do We Reproduce?

<!-- A great way to do this is to provide your configuration via a GitHub repository -->
<!-- The most helpful is a minimal reproduction with instructions on how to reproduce -->
<!-- Repositories with too many files or large `webpack.config.js` files are not suitable -->
<!-- Please only add small code snippets directly into this issue -->
<!-- https://gist.github.com is a good place for longer code snippets -->
<!-- If your issue is caused by a plugin or loader, please create an issue on the loader/plugin repository instead -->

I'm not sure, I feel like a simple repository using `webpack-dev-server` without any package loading `anymatch` and testing types.

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information
```
  System:
    OS: Linux 5.13 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
    Memory: 287.63 MB / 15.30 GB
  Binaries:
    Node: 17.3.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.6.0 - /usr/app/node_modules/.bin/npm
  Packages:
    babel-loader: ^8.x.x => 8.2.3 
    css-loader: ^6.x.x => 6.6.0 
    eslint-import-resolver-webpack: ^0.13.x => 0.13.2 
    eslint-loader: ^4.x.x => 4.0.2 
    html-webpack-plugin: ^5.x.x => 5.5.0 
    img-loader: ^3.x.x => 3.0.2 
    json-loader: ^0.5.x => 0.5.7 
    null-loader: ^4.x.x => 4.0.1 
    react-hot-loader: ^4.x.x => 4.13.0 
    style-loader: ^3.x.x => 3.3.1 
    svg-transform-loader: ^2.x.x => 2.0.13 
    terser-webpack-plugin: ^5.x.x => 5.3.1 
    webpack: ^5.x.x => 5.68.0 
    webpack-cli: ^4.x.x => 4.9.2 
    webpack-dev-server: >=4.7.1 => 4.7.4 
```

--- a/lib/Server.js
+++ b/lib/Server.js
@@ -25,7 +25,6 @@ const schema = require("./options.json");
 /** @typedef {import("express").NextFunction} NextFunction */
 /** @typedef {import("express").RequestHandler} ExpressRequestHandler */
 /** @typedef {import("express").ErrorRequestHandler} ExpressErrorRequestHandler */
-/** @typedef {import("anymatch").Matcher} AnymatchMatcher */
 /** @typedef {import("chokidar").WatchOptions} WatchOptions */
 /** @typedef {import("chokidar").FSWatcher} FSWatcher */
 /** @typedef {import("connect-history-api-fallback").Options} ConnectHistoryApiFallbackOptions */
@@ -65,7 +64,7 @@ const schema = require("./options.json");
 /**
  * @typedef {Object} WatchFiles
  * @property {string | string[]} paths
- * @property {WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} [options]
+ * @property {WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [options]
  */
 
 /**
@@ -74,7 +73,7 @@ const schema = require("./options.json");
  * @property {string | string[]} [publicPath]
  * @property {boolean | ServeIndexOptions} [serveIndex]
  * @property {ServeStaticOptions} [staticOptions]
- * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} [watch]
+ * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [watch]
  */
 
 /**
@@ -755,7 +754,7 @@ class Server {
     // TODO remove `{}` after drop webpack v4 support
     const compilerWatchOptions = compilerOptions.watchOptions || {};
     /**
-     * @param {WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} watchOptions
+     * @param {WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} watchOptions
      * @returns {WatchOptions}
      */
     const getWatchOptions = (watchOptions = {}) => {

--- a/types/lib/Server.d.ts
+++ b/types/lib/Server.d.ts
@@ -26,7 +26,6 @@ declare class Server {
         /** @typedef {import("express").NextFunction} NextFunction */
         /** @typedef {import("express").RequestHandler} ExpressRequestHandler */
         /** @typedef {import("express").ErrorRequestHandler} ExpressErrorRequestHandler */
-        /** @typedef {import("anymatch").Matcher} AnymatchMatcher */
         /** @typedef {import("chokidar").WatchOptions} WatchOptions */
         /** @typedef {import("chokidar").FSWatcher} FSWatcher */
         /** @typedef {import("connect-history-api-fallback").Options} ConnectHistoryApiFallbackOptions */
@@ -60,7 +59,7 @@ declare class Server {
         /**
          * @typedef {Object} WatchFiles
          * @property {string | string[]} paths
-         * @property {WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} [options]
+         * @property {WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [options]
          */
         /**
          * @typedef {Object} Static
@@ -68,7 +67,7 @@ declare class Server {
          * @property {string | string[]} [publicPath]
          * @property {boolean | ServeIndexOptions} [serveIndex]
          * @property {ServeStaticOptions} [staticOptions]
-         * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} [watch]
+         * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [watch]
          */
         /**
          * @typedef {Object} NormalizedStatic
@@ -221,6 +220,144 @@ declare class Server {
         simpleType: string;
       };
       "client-logging": {
+        /**
+         * @template Request, Response
+         * @typedef {import("webpack-dev-middleware").Options<Request, Response>} DevMiddlewareOptions
+         */
+        /**
+         * @template Request, Response
+         * @typedef {import("webpack-dev-middleware").Context<Request, Response>} DevMiddlewareContext
+         */
+        /**
+         * @typedef {"local-ip" | "local-ipv4" | "local-ipv6" | string} Host
+         */
+        /**
+         * @typedef {number | string | "auto"} Port
+         */
+        /**
+         * @typedef {Object} WatchFiles
+         * @property {string | string[]} paths
+         * @property {WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [options]
+         */
+        /**
+         * @typedef {Object} Static
+         * @property {string} [directory]
+         * @property {string | string[]} [publicPath]
+         * @property {boolean | ServeIndexOptions} [serveIndex]
+         * @property {ServeStaticOptions} [staticOptions]
+         * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [watch]
+         */
+        /**
+         * @typedef {Object} NormalizedStatic
+         * @property {string} directory
+         * @property {string[]} publicPath
+         * @property {false | ServeIndexOptions} serveIndex
+         * @property {ServeStaticOptions} staticOptions
+         * @property {false | WatchOptions} watch
+         */
+        /**
+         * @typedef {Object} ServerConfiguration
+         * @property {"http" | "https" | "spdy" | string} [type]
+         * @property {ServerOptions} [options]
+         */
+        /**
+         * @typedef {Object} WebSocketServerConfiguration
+         * @property {"sockjs" | "ws" | string | Function} [type]
+         * @property {Record<string, any>} [options]
+         */
+        /**
+         * @typedef {(import("ws").WebSocket | import("sockjs").Connection & { send: import("ws").WebSocket["send"], terminate: import("ws").WebSocket["terminate"], ping: import("ws").WebSocket["ping"] }) & { isAlive?: boolean }} ClientConnection
+         */
+        /**
+         * @typedef {import("ws").WebSocketServer | import("sockjs").Server & { close: import("ws").WebSocketServer["close"] }} WebSocketServer
+         */
+        /**
+         * @typedef {{ implementation: WebSocketServer, clients: ClientConnection[] }} WebSocketServerImplementation
+         */
+        /**
+         * @callback ByPass
+         * @param {Request} req
+         * @param {Response} res
+         * @param {ProxyConfigArrayItem} proxyConfig
+         */
+        /**
+         * @typedef {{ path?: HttpProxyMiddlewareOptionsFilter | undefined, context?: HttpProxyMiddlewareOptionsFilter | undefined } & { bypass?: ByPass } & HttpProxyMiddlewareOptions } ProxyConfigArrayItem
+         */
+        /**
+         * @typedef {(ProxyConfigArrayItem | ((req?: Request | undefined, res?: Response | undefined, next?: NextFunction | undefined) => ProxyConfigArrayItem))[]} ProxyConfigArray
+         */
+        /**
+         * @typedef {{ [url: string]: string | ProxyConfigArrayItem }} ProxyConfigMap
+         */
+        /**
+         * @typedef {Object} OpenApp
+         * @property {string} [name]
+         * @property {string[]} [arguments]
+         */
+        /**
+         * @typedef {Object} Open
+         * @property {string | string[] | OpenApp} [app]
+         * @property {string | string[]} [target]
+         */
+        /**
+         * @typedef {Object} NormalizedOpen
+         * @property {string} target
+         * @property {import("open").Options} options
+         */
+        /**
+         * @typedef {Object} WebSocketURL
+         * @property {string} [hostname]
+         * @property {string} [password]
+         * @property {string} [pathname]
+         * @property {number | string} [port]
+         * @property {string} [protocol]
+         * @property {string} [username]
+         */
+        /**
+         * @typedef {Object} ClientConfiguration
+         * @property {"log" | "info" | "warn" | "error" | "none" | "verbose"} [logging]
+         * @property {boolean  | { warnings?: boolean, errors?: boolean }} [overlay]
+         * @property {boolean} [progress]
+         * @property {boolean | number} [reconnect]
+         * @property {"ws" | "sockjs" | string} [webSocketTransport]
+         * @property {string | WebSocketURL} [webSocketURL]
+         */
+        /**
+         * @typedef {Array<{ key: string; value: string }> | Record<string, string | string[]>} Headers
+         */
+        /**
+         * @typedef {{ name?: string, path?: string, middleware: ExpressRequestHandler | ExpressErrorRequestHandler } | ExpressRequestHandler | ExpressErrorRequestHandler} Middleware
+         */
+        /**
+         * @typedef {Object} Configuration
+         * @property {boolean | string} [ipc]
+         * @property {Host} [host]
+         * @property {Port} [port]
+         * @property {boolean | "only"} [hot]
+         * @property {boolean} [liveReload]
+         * @property {DevMiddlewareOptions<Request, Response>} [devMiddleware]
+         * @property {boolean} [compress]
+         * @property {boolean} [magicHtml]
+         * @property {"auto" | "all" | string | string[]} [allowedHosts]
+         * @property {boolean | ConnectHistoryApiFallbackOptions} [historyApiFallback]
+         * @property {boolean} [setupExitSignals]
+         * @property {boolean | Record<string, never> | BonjourOptions} [bonjour]
+         * @property {string | string[] | WatchFiles | Array<string | WatchFiles>} [watchFiles]
+         * @property {boolean | string | Static | Array<string | Static>} [static]
+         * @property {boolean | ServerOptions} [https]
+         * @property {boolean} [http2]
+         * @property {"http" | "https" | "spdy" | string | ServerConfiguration} [server]
+         * @property {boolean | "sockjs" | "ws" | string | WebSocketServerConfiguration} [webSocketServer]
+         * @property {ProxyConfigMap | ProxyConfigArrayItem | ProxyConfigArray} [proxy]
+         * @property {boolean | string | Open | Array<string | Open>} [open]
+         * @property {boolean} [setupExitSignals]
+         * @property {boolean | ClientConfiguration} [client]
+         * @property {Headers | ((req: Request, res: Response, context: DevMiddlewareContext<Request, Response>) => Headers)} [headers]
+         * @property {(devServer: Server) => void} [onAfterSetupMiddleware]
+         * @property {(devServer: Server) => void} [onBeforeSetupMiddleware]
+         * @property {(devServer: Server) => void} [onListening]
+         * @property {(middlewares: Middleware[], devServer: Server) => Middleware[]} [setupMiddlewares]
+         */
         configs: {
           type: string;
           values: string[];
@@ -234,133 +371,6 @@ declare class Server {
       };
       "client-overlay": {
         configs: {
-          /**
-           * @typedef {number | string | "auto"} Port
-           */
-          /**
-           * @typedef {Object} WatchFiles
-           * @property {string | string[]} paths
-           * @property {WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} [options]
-           */
-          /**
-           * @typedef {Object} Static
-           * @property {string} [directory]
-           * @property {string | string[]} [publicPath]
-           * @property {boolean | ServeIndexOptions} [serveIndex]
-           * @property {ServeStaticOptions} [staticOptions]
-           * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: AnymatchMatcher | string[], poll?: number | boolean }} [watch]
-           */
-          /**
-           * @typedef {Object} NormalizedStatic
-           * @property {string} directory
-           * @property {string[]} publicPath
-           * @property {false | ServeIndexOptions} serveIndex
-           * @property {ServeStaticOptions} staticOptions
-           * @property {false | WatchOptions} watch
-           */
-          /**
-           * @typedef {Object} ServerConfiguration
-           * @property {"http" | "https" | "spdy" | string} [type]
-           * @property {ServerOptions} [options]
-           */
-          /**
-           * @typedef {Object} WebSocketServerConfiguration
-           * @property {"sockjs" | "ws" | string | Function} [type]
-           * @property {Record<string, any>} [options]
-           */
-          /**
-           * @typedef {(import("ws").WebSocket | import("sockjs").Connection & { send: import("ws").WebSocket["send"], terminate: import("ws").WebSocket["terminate"], ping: import("ws").WebSocket["ping"] }) & { isAlive?: boolean }} ClientConnection
-           */
-          /**
-           * @typedef {import("ws").WebSocketServer | import("sockjs").Server & { close: import("ws").WebSocketServer["close"] }} WebSocketServer
-           */
-          /**
-           * @typedef {{ implementation: WebSocketServer, clients: ClientConnection[] }} WebSocketServerImplementation
-           */
-          /**
-           * @callback ByPass
-           * @param {Request} req
-           * @param {Response} res
-           * @param {ProxyConfigArrayItem} proxyConfig
-           */
-          /**
-           * @typedef {{ path?: HttpProxyMiddlewareOptionsFilter | undefined, context?: HttpProxyMiddlewareOptionsFilter | undefined } & { bypass?: ByPass } & HttpProxyMiddlewareOptions } ProxyConfigArrayItem
-           */
-          /**
-           * @typedef {(ProxyConfigArrayItem | ((req?: Request | undefined, res?: Response | undefined, next?: NextFunction | undefined) => ProxyConfigArrayItem))[]} ProxyConfigArray
-           */
-          /**
-           * @typedef {{ [url: string]: string | ProxyConfigArrayItem }} ProxyConfigMap
-           */
-          /**
-           * @typedef {Object} OpenApp
-           * @property {string} [name]
-           * @property {string[]} [arguments]
-           */
-          /**
-           * @typedef {Object} Open
-           * @property {string | string[] | OpenApp} [app]
-           * @property {string | string[]} [target]
-           */
-          /**
-           * @typedef {Object} NormalizedOpen
-           * @property {string} target
-           * @property {import("open").Options} options
-           */
-          /**
-           * @typedef {Object} WebSocketURL
-           * @property {string} [hostname]
-           * @property {string} [password]
-           * @property {string} [pathname]
-           * @property {number | string} [port]
-           * @property {string} [protocol]
-           * @property {string} [username]
-           */
-          /**
-           * @typedef {Object} ClientConfiguration
-           * @property {"log" | "info" | "warn" | "error" | "none" | "verbose"} [logging]
-           * @property {boolean  | { warnings?: boolean, errors?: boolean }} [overlay]
-           * @property {boolean} [progress]
-           * @property {boolean | number} [reconnect]
-           * @property {"ws" | "sockjs" | string} [webSocketTransport]
-           * @property {string | WebSocketURL} [webSocketURL]
-           */
-          /**
-           * @typedef {Array<{ key: string; value: string }> | Record<string, string | string[]>} Headers
-           */
-          /**
-           * @typedef {{ name?: string, path?: string, middleware: ExpressRequestHandler | ExpressErrorRequestHandler } | ExpressRequestHandler | ExpressErrorRequestHandler} Middleware
-           */
-          /**
-           * @typedef {Object} Configuration
-           * @property {boolean | string} [ipc]
-           * @property {Host} [host]
-           * @property {Port} [port]
-           * @property {boolean | "only"} [hot]
-           * @property {boolean} [liveReload]
-           * @property {DevMiddlewareOptions<Request, Response>} [devMiddleware]
-           * @property {boolean} [compress]
-           * @property {boolean} [magicHtml]
-           * @property {"auto" | "all" | string | string[]} [allowedHosts]
-           * @property {boolean | ConnectHistoryApiFallbackOptions} [historyApiFallback]
-           * @property {boolean} [setupExitSignals]
-           * @property {boolean | Record<string, never> | BonjourOptions} [bonjour]
-           * @property {string | string[] | WatchFiles | Array<string | WatchFiles>} [watchFiles]
-           * @property {boolean | string | Static | Array<string | Static>} [static]
-           * @property {boolean | ServerOptions} [https]
-           * @property {boolean} [http2]
-           * @property {"http" | "https" | "spdy" | string | ServerConfiguration} [server]
-           * @property {boolean | "sockjs" | "ws" | string | WebSocketServerConfiguration} [webSocketServer]
-           * @property {ProxyConfigMap | ProxyConfigArrayItem | ProxyConfigArray} [proxy]
-           * @property {boolean | string | Open | Array<string | Open>} [open]
-           * @property {boolean} [setupExitSignals]
-           * @property {boolean | ClientConfiguration} [client]
-           * @property {Headers | ((req: Request, res: Response, context: DevMiddlewareContext<Request, Response>) => Headers)} [headers]
-           * @property {(devServer: Server) => void} [onAfterSetupMiddleware]
-           * @property {(devServer: Server) => void} [onBeforeSetupMiddleware]
-           * @property {(devServer: Server) => void} [onListening]
-           * @property {(middlewares: Middleware[], devServer: Server) => Middleware[]} [setupMiddlewares]
-           */
           type: string;
           multiple: boolean;
           description: string;
@@ -443,6 +453,36 @@ declare class Server {
               path: string;
             }
         )[];
+        /**
+         * @typedef {Object} Configuration
+         * @property {boolean | string} [ipc]
+         * @property {Host} [host]
+         * @property {Port} [port]
+         * @property {boolean | "only"} [hot]
+         * @property {boolean} [liveReload]
+         * @property {DevMiddlewareOptions<Request, Response>} [devMiddleware]
+         * @property {boolean} [compress]
+         * @property {boolean} [magicHtml]
+         * @property {"auto" | "all" | string | string[]} [allowedHosts]
+         * @property {boolean | ConnectHistoryApiFallbackOptions} [historyApiFallback]
+         * @property {boolean} [setupExitSignals]
+         * @property {boolean | Record<string, never> | BonjourOptions} [bonjour]
+         * @property {string | string[] | WatchFiles | Array<string | WatchFiles>} [watchFiles]
+         * @property {boolean | string | Static | Array<string | Static>} [static]
+         * @property {boolean | ServerOptions} [https]
+         * @property {boolean} [http2]
+         * @property {"http" | "https" | "spdy" | string | ServerConfiguration} [server]
+         * @property {boolean | "sockjs" | "ws" | string | WebSocketServerConfiguration} [webSocketServer]
+         * @property {ProxyConfigMap | ProxyConfigArrayItem | ProxyConfigArray} [proxy]
+         * @property {boolean | string | Open | Array<string | Open>} [open]
+         * @property {boolean} [setupExitSignals]
+         * @property {boolean | ClientConfiguration} [client]
+         * @property {Headers | ((req: Request, res: Response, context: DevMiddlewareContext<Request, Response>) => Headers)} [headers]
+         * @property {(devServer: Server) => void} [onAfterSetupMiddleware]
+         * @property {(devServer: Server) => void} [onBeforeSetupMiddleware]
+         * @property {(devServer: Server) => void} [onListening]
+         * @property {(middlewares: Middleware[], devServer: Server) => Middleware[]} [setupMiddlewares]
+         */
         description: string;
         simpleType: string;
         multiple: boolean;
@@ -508,7 +548,7 @@ declare class Server {
               description: string;
               multiple: boolean;
               path: string;
-              /** @type {Compiler | MultiCompiler} */ type: string;
+              type: string;
               values: string[];
             }
           | {
@@ -518,6 +558,10 @@ declare class Server {
               type: string;
             }
         )[];
+        /**
+         * @private
+         * @type {{ name: string | symbol, listener: (...args: any[]) => void}[] }}
+         */
         description: string;
         multiple: boolean;
         simpleType: string;
@@ -531,15 +575,8 @@ declare class Server {
           type: string;
           multiple: boolean;
           description: string;
-          /**
-           * @type {Socket[]}
-           */
           path: string;
         }[];
-        /**
-         * @private
-         * @type {string | undefined}
-         */
         description: string;
         simpleType: string;
         multiple: boolean;
@@ -595,6 +632,10 @@ declare class Server {
               multiple: boolean;
               description: string;
               negatedDescription: string;
+              /**
+               * @param {"v4" | "v6"} family
+               * @returns {string | undefined}
+               */
               path: string;
             }
           | {
@@ -621,6 +662,10 @@ declare class Server {
         simpleType: string;
         multiple: boolean;
       };
+      /**
+       * @param {Port} port
+       * @returns {Promise<number | string>}
+       */
       https: {
         configs: {
           type: string;
@@ -640,6 +685,9 @@ declare class Server {
           description: string;
           path: string;
         }[];
+        /**
+         * @type {string | undefined}
+         */
         description: string;
         simpleType: string;
         multiple: boolean;
@@ -671,11 +719,10 @@ declare class Server {
           description: string;
           multiple: boolean;
           path: string;
-          type: string;
+          /** @type {WebSocketURL} */ type: string;
         }[];
         description: string;
         multiple: boolean;
-        /** @type {{ type: WebSocketServerConfiguration["type"], options: NonNullable<WebSocketServerConfiguration["options"]> }} */
         simpleType: string;
       };
       "https-cert": {
@@ -689,7 +736,6 @@ declare class Server {
         simpleType: string;
         multiple: boolean;
       };
-      /** @type {ServerConfiguration} */
       "https-cert-reset": {
         configs: {
           description: string;
@@ -716,7 +762,7 @@ declare class Server {
         configs: {
           description: string;
           multiple: boolean;
-          /** @type {number | string} */ path: string;
+          path: string;
           type: string;
         }[];
         description: string;
@@ -738,7 +784,7 @@ declare class Server {
         configs: {
           description: string;
           multiple: boolean;
-          /** @type {string} */ path: string;
+          path: string;
           type: string;
         }[];
         description: string;
@@ -800,7 +846,7 @@ declare class Server {
             }
           | {
               type: string;
-              values: boolean[];
+              /** @type {Object<string,string>} */ values: boolean[];
               multiple: boolean;
               description: string;
               path: string;
@@ -831,7 +877,7 @@ declare class Server {
           path: string;
         }[];
         description: string;
-        /** @type {MultiCompiler} */ simpleType: string;
+        simpleType: string;
         multiple: boolean;
       };
       open: {
@@ -865,10 +911,6 @@ declare class Server {
         simpleType: string;
         multiple: boolean;
       };
-      /**
-       * @private
-       * @returns {Promise<void>}
-       */
       "open-app-name": {
         configs: {
           type: string;
@@ -928,7 +970,7 @@ declare class Server {
         configs: (
           | {
               type: string;
-              /** @type {NormalizedStatic} */ multiple: boolean;
+              multiple: boolean;
               description: string;
               path: string;
             }
@@ -1109,9 +1151,8 @@ declare class Server {
         }[];
         description: string;
         multiple: boolean;
-        simpleType: string /** @type {Array<keyof ServerOptions>} */;
+        simpleType: string;
       };
-      /** @type {Array<keyof ServerOptions>} */
       static: {
         configs: (
           | {
@@ -1129,10 +1170,9 @@ declare class Server {
             }
         )[];
         description: string;
-        /** @type {any} */ simpleType: string;
-        multiple: boolean /** @type {ServerOptions} */;
+        simpleType: string;
+        multiple: boolean;
       };
-      /** @type {ServerOptions} */
       "static-directory": {
         configs: {
           type: string;
@@ -1159,7 +1199,7 @@ declare class Server {
         configs: {
           type: string;
           multiple: boolean;
-          /** @type {ServerOptions} */ description: string;
+          description: string;
           path: string;
         }[];
         description: string;
@@ -1244,7 +1284,6 @@ declare class Server {
               description: string;
               multiple: boolean;
               path: string;
-              /** @type {ServerOptions & { cacert?: ServerOptions["ca"] }} */
               type: string;
             }
         )[];
@@ -1363,6 +1402,151 @@ declare class Server {
                 logging: {
                   $ref: string;
                 };
+                /** @typedef {import("serve-static").ServeStaticOptions} ServeStaticOptions */
+                /** @typedef {import("ipaddr.js").IPv4} IPv4 */
+                /** @typedef {import("ipaddr.js").IPv6} IPv6 */
+                /** @typedef {import("net").Socket} Socket */
+                /** @typedef {import("http").IncomingMessage} IncomingMessage */
+                /** @typedef {import("open").Options} OpenOptions */
+                /** @typedef {import("https").ServerOptions & { spdy?: { plain?: boolean | undefined, ssl?: boolean | undefined, 'x-forwarded-for'?: string | undefined, protocol?: string | undefined, protocols?: string[] | undefined }}} ServerOptions */
+                /**
+                 * @template Request, Response
+                 * @typedef {import("webpack-dev-middleware").Options<Request, Response>} DevMiddlewareOptions
+                 */
+                /**
+                 * @template Request, Response
+                 * @typedef {import("webpack-dev-middleware").Context<Request, Response>} DevMiddlewareContext
+                 */
+                /**
+                 * @typedef {"local-ip" | "local-ipv4" | "local-ipv6" | string} Host
+                 */
+                /**
+                 * @typedef {number | string | "auto"} Port
+                 */
+                /**
+                 * @typedef {Object} WatchFiles
+                 * @property {string | string[]} paths
+                 * @property {WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [options]
+                 */
+                /**
+                 * @typedef {Object} Static
+                 * @property {string} [directory]
+                 * @property {string | string[]} [publicPath]
+                 * @property {boolean | ServeIndexOptions} [serveIndex]
+                 * @property {ServeStaticOptions} [staticOptions]
+                 * @property {boolean | WatchOptions & { aggregateTimeout?: number, ignored?: WatchOptions["ignored"], poll?: number | boolean }} [watch]
+                 */
+                /**
+                 * @typedef {Object} NormalizedStatic
+                 * @property {string} directory
+                 * @property {string[]} publicPath
+                 * @property {false | ServeIndexOptions} serveIndex
+                 * @property {ServeStaticOptions} staticOptions
+                 * @property {false | WatchOptions} watch
+                 */
+                /**
+                 * @typedef {Object} ServerConfiguration
+                 * @property {"http" | "https" | "spdy" | string} [type]
+                 * @property {ServerOptions} [options]
+                 */
+                /**
+                 * @typedef {Object} WebSocketServerConfiguration
+                 * @property {"sockjs" | "ws" | string | Function} [type]
+                 * @property {Record<string, any>} [options]
+                 */
+                /**
+                 * @typedef {(import("ws").WebSocket | import("sockjs").Connection & { send: import("ws").WebSocket["send"], terminate: import("ws").WebSocket["terminate"], ping: import("ws").WebSocket["ping"] }) & { isAlive?: boolean }} ClientConnection
+                 */
+                /**
+                 * @typedef {import("ws").WebSocketServer | import("sockjs").Server & { close: import("ws").WebSocketServer["close"] }} WebSocketServer
+                 */
+                /**
+                 * @typedef {{ implementation: WebSocketServer, clients: ClientConnection[] }} WebSocketServerImplementation
+                 */
+                /**
+                 * @callback ByPass
+                 * @param {Request} req
+                 * @param {Response} res
+                 * @param {ProxyConfigArrayItem} proxyConfig
+                 */
+                /**
+                 * @typedef {{ path?: HttpProxyMiddlewareOptionsFilter | undefined, context?: HttpProxyMiddlewareOptionsFilter | undefined } & { bypass?: ByPass } & HttpProxyMiddlewareOptions } ProxyConfigArrayItem
+                 */
+                /**
+                 * @typedef {(ProxyConfigArrayItem | ((req?: Request | undefined, res?: Response | undefined, next?: NextFunction | undefined) => ProxyConfigArrayItem))[]} ProxyConfigArray
+                 */
+                /**
+                 * @typedef {{ [url: string]: string | ProxyConfigArrayItem }} ProxyConfigMap
+                 */
+                /**
+                 * @typedef {Object} OpenApp
+                 * @property {string} [name]
+                 * @property {string[]} [arguments]
+                 */
+                /**
+                 * @typedef {Object} Open
+                 * @property {string | string[] | OpenApp} [app]
+                 * @property {string | string[]} [target]
+                 */
+                /**
+                 * @typedef {Object} NormalizedOpen
+                 * @property {string} target
+                 * @property {import("open").Options} options
+                 */
+                /**
+                 * @typedef {Object} WebSocketURL
+                 * @property {string} [hostname]
+                 * @property {string} [password]
+                 * @property {string} [pathname]
+                 * @property {number | string} [port]
+                 * @property {string} [protocol]
+                 * @property {string} [username]
+                 */
+                /**
+                 * @typedef {Object} ClientConfiguration
+                 * @property {"log" | "info" | "warn" | "error" | "none" | "verbose"} [logging]
+                 * @property {boolean  | { warnings?: boolean, errors?: boolean }} [overlay]
+                 * @property {boolean} [progress]
+                 * @property {boolean | number} [reconnect]
+                 * @property {"ws" | "sockjs" | string} [webSocketTransport]
+                 * @property {string | WebSocketURL} [webSocketURL]
+                 */
+                /**
+                 * @typedef {Array<{ key: string; value: string }> | Record<string, string | string[]>} Headers
+                 */
+                /**
+                 * @typedef {{ name?: string, path?: string, middleware: ExpressRequestHandler | ExpressErrorRequestHandler } | ExpressRequestHandler | ExpressErrorRequestHandler} Middleware
+                 */
+                /**
+                 * @typedef {Object} Configuration
+                 * @property {boolean | string} [ipc]
+                 * @property {Host} [host]
+                 * @property {Port} [port]
+                 * @property {boolean | "only"} [hot]
+                 * @property {boolean} [liveReload]
+                 * @property {DevMiddlewareOptions<Request, Response>} [devMiddleware]
+                 * @property {boolean} [compress]
+                 * @property {boolean} [magicHtml]
+                 * @property {"auto" | "all" | string | string[]} [allowedHosts]
+                 * @property {boolean | ConnectHistoryApiFallbackOptions} [historyApiFallback]
+                 * @property {boolean} [setupExitSignals]
+                 * @property {boolean | Record<string, never> | BonjourOptions} [bonjour]
+                 * @property {string | string[] | WatchFiles | Array<string | WatchFiles>} [watchFiles]
+                 * @property {boolean | string | Static | Array<string | Static>} [static]
+                 * @property {boolean | ServerOptions} [https]
+                 * @property {boolean} [http2]
+                 * @property {"http" | "https" | "spdy" | string | ServerConfiguration} [server]
+                 * @property {boolean | "sockjs" | "ws" | string | WebSocketServerConfiguration} [webSocketServer]
+                 * @property {ProxyConfigMap | ProxyConfigArrayItem | ProxyConfigArray} [proxy]
+                 * @property {boolean | string | Open | Array<string | Open>} [open]
+                 * @property {boolean} [setupExitSignals]
+                 * @property {boolean | ClientConfiguration} [client]
+                 * @property {Headers | ((req: Request, res: Response, context: DevMiddlewareContext<Request, Response>) => Headers)} [headers]
+                 * @property {(devServer: Server) => void} [onAfterSetupMiddleware]
+                 * @property {(devServer: Server) => void} [onBeforeSetupMiddleware]
+                 * @property {(devServer: Server) => void} [onListening]
+                 * @property {(middlewares: Middleware[], devServer: Server) => Middleware[]} [setupMiddlewares]
+                 */
                 overlay: {
                   $ref: string;
                 };
@@ -1413,117 +1597,6 @@ declare class Server {
                   };
                 };
                 warnings: {
-                  /**
-                   * @typedef {Object} NormalizedStatic
-                   * @property {string} directory
-                   * @property {string[]} publicPath
-                   * @property {false | ServeIndexOptions} serveIndex
-                   * @property {ServeStaticOptions} staticOptions
-                   * @property {false | WatchOptions} watch
-                   */
-                  /**
-                   * @typedef {Object} ServerConfiguration
-                   * @property {"http" | "https" | "spdy" | string} [type]
-                   * @property {ServerOptions} [options]
-                   */
-                  /**
-                   * @typedef {Object} WebSocketServerConfiguration
-                   * @property {"sockjs" | "ws" | string | Function} [type]
-                   * @property {Record<string, any>} [options]
-                   */
-                  /**
-                   * @typedef {(import("ws").WebSocket | import("sockjs").Connection & { send: import("ws").WebSocket["send"], terminate: import("ws").WebSocket["terminate"], ping: import("ws").WebSocket["ping"] }) & { isAlive?: boolean }} ClientConnection
-                   */
-                  /**
-                   * @typedef {import("ws").WebSocketServer | import("sockjs").Server & { close: import("ws").WebSocketServer["close"] }} WebSocketServer
-                   */
-                  /**
-                   * @typedef {{ implementation: WebSocketServer, clients: ClientConnection[] }} WebSocketServerImplementation
-                   */
-                  /**
-                   * @callback ByPass
-                   * @param {Request} req
-                   * @param {Response} res
-                   * @param {ProxyConfigArrayItem} proxyConfig
-                   */
-                  /**
-                   * @typedef {{ path?: HttpProxyMiddlewareOptionsFilter | undefined, context?: HttpProxyMiddlewareOptionsFilter | undefined } & { bypass?: ByPass } & HttpProxyMiddlewareOptions } ProxyConfigArrayItem
-                   */
-                  /**
-                   * @typedef {(ProxyConfigArrayItem | ((req?: Request | undefined, res?: Response | undefined, next?: NextFunction | undefined) => ProxyConfigArrayItem))[]} ProxyConfigArray
-                   */
-                  /**
-                   * @typedef {{ [url: string]: string | ProxyConfigArrayItem }} ProxyConfigMap
-                   */
-                  /**
-                   * @typedef {Object} OpenApp
-                   * @property {string} [name]
-                   * @property {string[]} [arguments]
-                   */
-                  /**
-                   * @typedef {Object} Open
-                   * @property {string | string[] | OpenApp} [app]
-                   * @property {string | string[]} [target]
-                   */
-                  /**
-                   * @typedef {Object} NormalizedOpen
-                   * @property {string} target
-                   * @property {import("open").Options} options
-                   */
-                  /**
-                   * @typedef {Object} WebSocketURL
-                   * @property {string} [hostname]
-                   * @property {string} [password]
-                   * @property {string} [pathname]
-                   * @property {number | string} [port]
-                   * @property {string} [protocol]
-                   * @property {string} [username]
-                   */
-                  /**
-                   * @typedef {Object} ClientConfiguration
-                   * @property {"log" | "info" | "warn" | "error" | "none" | "verbose"} [logging]
-                   * @property {boolean  | { warnings?: boolean, errors?: boolean }} [overlay]
-                   * @property {boolean} [progress]
-                   * @property {boolean | number} [reconnect]
-                   * @property {"ws" | "sockjs" | string} [webSocketTransport]
-                   * @property {string | WebSocketURL} [webSocketURL]
-                   */
-                  /**
-                   * @typedef {Array<{ key: string; value: string }> | Record<string, string | string[]>} Headers
-                   */
-                  /**
-                   * @typedef {{ name?: string, path?: string, middleware: ExpressRequestHandler | ExpressErrorRequestHandler } | ExpressRequestHandler | ExpressErrorRequestHandler} Middleware
-                   */
-                  /**
-                   * @typedef {Object} Configuration
-                   * @property {boolean | string} [ipc]
-                   * @property {Host} [host]
-                   * @property {Port} [port]
-                   * @property {boolean | "only"} [hot]
-                   * @property {boolean} [liveReload]
-                   * @property {DevMiddlewareOptions<Request, Response>} [devMiddleware]
-                   * @property {boolean} [compress]
-                   * @property {boolean} [magicHtml]
-                   * @property {"auto" | "all" | string | string[]} [allowedHosts]
-                   * @property {boolean | ConnectHistoryApiFallbackOptions} [historyApiFallback]
-                   * @property {boolean} [setupExitSignals]
-                   * @property {boolean | Record<string, never> | BonjourOptions} [bonjour]
-                   * @property {string | string[] | WatchFiles | Array<string | WatchFiles>} [watchFiles]
-                   * @property {boolean | string | Static | Array<string | Static>} [static]
-                   * @property {boolean | ServerOptions} [https]
-                   * @property {boolean} [http2]
-                   * @property {"http" | "https" | "spdy" | string | ServerConfiguration} [server]
-                   * @property {boolean | "sockjs" | "ws" | string | WebSocketServerConfiguration} [webSocketServer]
-                   * @property {ProxyConfigMap | ProxyConfigArrayItem | ProxyConfigArray} [proxy]
-                   * @property {boolean | string | Open | Array<string | Open>} [open]
-                   * @property {boolean} [setupExitSignals]
-                   * @property {boolean | ClientConfiguration} [client]
-                   * @property {Headers | ((req: Request, res: Response, context: DevMiddlewareContext<Request, Response>) => Headers)} [headers]
-                   * @property {(devServer: Server) => void} [onAfterSetupMiddleware]
-                   * @property {(devServer: Server) => void} [onBeforeSetupMiddleware]
-                   * @property {(devServer: Server) => void} [onListening]
-                   * @property {(middlewares: Middleware[], devServer: Server) => Middleware[]} [setupMiddlewares]
-                   */
                   description: string;
                   type: string;
                   cli: {
@@ -1548,6 +1621,90 @@ declare class Server {
       ClientReconnect: {
         description: string;
         link: string;
+        /**
+         * @callback ByPass
+         * @param {Request} req
+         * @param {Response} res
+         * @param {ProxyConfigArrayItem} proxyConfig
+         */
+        /**
+         * @typedef {{ path?: HttpProxyMiddlewareOptionsFilter | undefined, context?: HttpProxyMiddlewareOptionsFilter | undefined } & { bypass?: ByPass } & HttpProxyMiddlewareOptions } ProxyConfigArrayItem
+         */
+        /**
+         * @typedef {(ProxyConfigArrayItem | ((req?: Request | undefined, res?: Response | undefined, next?: NextFunction | undefined) => ProxyConfigArrayItem))[]} ProxyConfigArray
+         */
+        /**
+         * @typedef {{ [url: string]: string | ProxyConfigArrayItem }} ProxyConfigMap
+         */
+        /**
+         * @typedef {Object} OpenApp
+         * @property {string} [name]
+         * @property {string[]} [arguments]
+         */
+        /**
+         * @typedef {Object} Open
+         * @property {string | string[] | OpenApp} [app]
+         * @property {string | string[]} [target]
+         */
+        /**
+         * @typedef {Object} NormalizedOpen
+         * @property {string} target
+         * @property {import("open").Options} options
+         */
+        /**
+         * @typedef {Object} WebSocketURL
+         * @property {string} [hostname]
+         * @property {string} [password]
+         * @property {string} [pathname]
+         * @property {number | string} [port]
+         * @property {string} [protocol]
+         * @property {string} [username]
+         */
+        /**
+         * @typedef {Object} ClientConfiguration
+         * @property {"log" | "info" | "warn" | "error" | "none" | "verbose"} [logging]
+         * @property {boolean  | { warnings?: boolean, errors?: boolean }} [overlay]
+         * @property {boolean} [progress]
+         * @property {boolean | number} [reconnect]
+         * @property {"ws" | "sockjs" | string} [webSocketTransport]
+         * @property {string | WebSocketURL} [webSocketURL]
+         */
+        /**
+         * @typedef {Array<{ key: string; value: string }> | Record<string, string | string[]>} Headers
+         */
+        /**
+         * @typedef {{ name?: string, path?: string, middleware: ExpressRequestHandler | ExpressErrorRequestHandler } | ExpressRequestHandler | ExpressErrorRequestHandler} Middleware
+         */
+        /**
+         * @typedef {Object} Configuration
+         * @property {boolean | string} [ipc]
+         * @property {Host} [host]
+         * @property {Port} [port]
+         * @property {boolean | "only"} [hot]
+         * @property {boolean} [liveReload]
+         * @property {DevMiddlewareOptions<Request, Response>} [devMiddleware]
+         * @property {boolean} [compress]
+         * @property {boolean} [magicHtml]
+         * @property {"auto" | "all" | string | string[]} [allowedHosts]
+         * @property {boolean | ConnectHistoryApiFallbackOptions} [historyApiFallback]
+         * @property {boolean} [setupExitSignals]
+         * @property {boolean | Record<string, never> | BonjourOptions} [bonjour]
+         * @property {string | string[] | WatchFiles | Array<string | WatchFiles>} [watchFiles]
+         * @property {boolean | string | Static | Array<string | Static>} [static]
+         * @property {boolean | ServerOptions} [https]
+         * @property {boolean} [http2]
+         * @property {"http" | "https" | "spdy" | string | ServerConfiguration} [server]
+         * @property {boolean | "sockjs" | "ws" | string | WebSocketServerConfiguration} [webSocketServer]
+         * @property {ProxyConfigMap | ProxyConfigArrayItem | ProxyConfigArray} [proxy]
+         * @property {boolean | string | Open | Array<string | Open>} [open]
+         * @property {boolean} [setupExitSignals]
+         * @property {boolean | ClientConfiguration} [client]
+         * @property {Headers | ((req: Request, res: Response, context: DevMiddlewareContext<Request, Response>) => Headers)} [headers]
+         * @property {(devServer: Server) => void} [onAfterSetupMiddleware]
+         * @property {(devServer: Server) => void} [onBeforeSetupMiddleware]
+         * @property {(devServer: Server) => void} [onListening]
+         * @property {(middlewares: Middleware[], devServer: Server) => Middleware[]} [setupMiddlewares]
+         */
         anyOf: (
           | {
               type: string;
@@ -1600,6 +1757,9 @@ declare class Server {
                   description: string;
                   type: string;
                 };
+                /**
+                 * @typedef {Array<{ key: string; value: string }> | Record<string, string | string[]>} Headers
+                 */
                 password: {
                   description: string;
                   type: string;
@@ -1713,6 +1873,10 @@ declare class Server {
                       }
                     | {
                         instanceof: string;
+                        /**
+                         * @private
+                         * @type {RequestHandler[]}
+                         */
                         type?: undefined;
                         items?: undefined;
                       }
@@ -1810,10 +1974,6 @@ declare class Server {
                         items?: undefined;
                       }
                   )[];
-                  /**
-                   * @param {"v4" | "v6"} family
-                   * @returns {string | undefined}
-                   */
                   description: string;
                 };
                 key: {
@@ -1919,9 +2079,6 @@ declare class Server {
       Headers: {
         anyOf: (
           | {
-              /**
-               * @type {string[]}
-               */
               type: string;
               items: {
                 $ref: string;
@@ -1957,13 +2114,13 @@ declare class Server {
             }
           | {
               type: string;
-              /** @type {WebSocketURL} */ description: string;
+              description: string;
               link: string;
               cli?: undefined /** @typedef {import("express").Request} Request */;
             }
         )[];
         description: string;
-        link: string;
+        /** @type {string} */ link: string;
       };
       Host: {
         description: string;
@@ -1971,7 +2128,7 @@ declare class Server {
         anyOf: (
           | {
               enum: string[];
-              /** @type {ServerConfiguration} */ type?: undefined;
+              type?: undefined;
               minLength?: undefined;
             }
           | {
@@ -2110,12 +2267,6 @@ declare class Server {
                           }
                         | {
                             type: string;
-                            /**
-                             * prependEntry Method for webpack 4
-                             * @param {any} originalEntry
-                             * @param {any} newAdditionalEntries
-                             * @returns {any}
-                             */
                             minLength: number;
                             items?: undefined;
                             minItems?: undefined;
@@ -2140,7 +2291,6 @@ declare class Server {
                   cli: {
                     description: string;
                   };
-                  /** @type {Object<string,string>} */
                   additionalProperties?: undefined;
                   properties?: undefined;
                 }
@@ -2196,7 +2346,7 @@ declare class Server {
                     }
                   | {
                       instanceof: string;
-                      type?: undefined;
+                      /** @type {string} */ type?: undefined;
                     }
                 )[];
               };
@@ -2207,7 +2357,7 @@ declare class Server {
       };
       Server: {
         anyOf: {
-          $ref: string;
+          $ref: string /** @type {MultiCompiler} */;
         }[];
         link: string;
         description: string;
@@ -2248,7 +2398,7 @@ declare class Server {
         properties: {
           passphrase: {
             type: string;
-            /** @type {MultiCompiler} */ description: string;
+            description: string;
           };
           requestCert: {
             type: string;
@@ -2375,7 +2525,7 @@ declare class Server {
                 }
               | {
                   instanceof: string;
-                  /** @type {NormalizedStatic} */ type?: undefined;
+                  type?: undefined;
                   items?: undefined;
                 }
             )[];
@@ -2641,7 +2791,7 @@ declare class Server {
           $ref: string;
         }[];
         description: string;
-        link: string /** @type {Array<keyof ServerOptions>} */;
+        link: string;
       };
       WebSocketServerType: {
         enum: string[];
@@ -2700,6 +2850,7 @@ declare class Server {
         $ref: string;
       };
       client: {
+        /** @type {any} */
         $ref: string;
       };
       compress: {
@@ -3098,7 +3249,6 @@ declare namespace Server {
     NextFunction,
     ExpressRequestHandler,
     ExpressErrorRequestHandler,
-    AnymatchMatcher,
     WatchOptions,
     FSWatcher,
     ConnectHistoryApiFallbackOptions,
@@ -3237,7 +3387,6 @@ type Response = import("express").Response;
 type NextFunction = import("express").NextFunction;
 type ExpressRequestHandler = import("express").RequestHandler;
 type ExpressErrorRequestHandler = import("express").ErrorRequestHandler;
-type AnymatchMatcher = import("anymatch").Matcher;
 type WatchOptions = import("chokidar").WatchOptions;
 type ConnectHistoryApiFallbackOptions =
   import("connect-history-api-fallback").Options;
@@ -3270,7 +3419,7 @@ type WatchFiles = {
   options?:
     | (import("chokidar").WatchOptions & {
         aggregateTimeout?: number | undefined;
-        ignored?: string[] | import("anymatch").Matcher | undefined;
+        ignored?: WatchOptions["ignored"];
         poll?: number | boolean | undefined;
       })
     | undefined;
@@ -3286,7 +3435,7 @@ type Static = {
     | boolean
     | (import("chokidar").WatchOptions & {
         aggregateTimeout?: number | undefined;
-        ignored?: string[] | import("anymatch").Matcher | undefined;
+        ignored?: WatchOptions["ignored"];
         poll?: number | boolean | undefined;
       })
     | undefined;

